core/chains/evm/txm/storage/inmemory_store_manager.go

```diff
@@ -0,0 +1,136 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"go.uber.org/multierr"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txm/types"
+)
+
+const StoreNotFoundForAddress string = "InMemoryStore for address: %v not found"
+
+type InMemoryStoreManager struct {
+	lggr             logger.Logger
+	chainID          *big.Int
+	InMemoryStoreMap map[common.Address]*InMemoryStore
+}
+
+func NewInMemoryStoreManager(lggr logger.Logger, chainID *big.Int) *InMemoryStoreManager {
+	inMemoryStoreMap := make(map[common.Address]*InMemoryStore)
+	return &InMemoryStoreManager{
+		lggr:             lggr,
+		chainID:          chainID,
+		InMemoryStoreMap: inMemoryStoreMap}
+}
+
+func (m *InMemoryStoreManager) AbandonPendingTransactions(_ context.Context, fromAddress common.Address) error {
+	if store, exists := m.InMemoryStoreMap[fromAddress]; exists {
+		store.AbandonPendingTransactions()
+		return nil
+	}
+	return fmt.Errorf(StoreNotFoundForAddress, fromAddress)
+}
+
+func (m *InMemoryStoreManager) Add(addresses ...common.Address) (err error) {
+	for _, address := range addresses {
+		if _, exists := m.InMemoryStoreMap[address]; exists {
+			err = multierr.Append(err, fmt.Errorf("address %v already exists in store manager", address))
```

Comment:
nit: do we need to use multierr here? Shouldn't we just use the native error wrapping in `fmt.Errorf`?